packages/cli/test/generate-index.spec.ts

```diff
@@ -111,6 +111,52 @@ describe('build index', () => {
             ].join('\n')
         );
     });
+
+    it('should create index file using a custom generator with named exports generation and @namespace', () => {
+        const fs = createFS({
+            '/comp-A.st.css': `
+                :vars {
+                    color1: red;
+                }
+                .a{
+                    --color2: red;
+                }
+                @keyframes X {}
+            `,
+            '/b/1-some-comp-B-.st.css': `
+               .b{}
+            `,
+        });
+
+        const stylable = new Stylable('/', fs, () => ({}));
+
+        build({
+            extension: '.st.css',
+            fs,
+            stylable,
+            outDir: '.',
+            srcDir: '.',
+            indexFile: 'index.st.css',
+            rootDir: resolve('/'),
+            log,
+            generatorPath: require.resolve('./fixtures/named-exports-generator.ts'),
+        });
+
+        const res = fs.readFileSync(resolve('/index.st.css')).toString();
+
+        expect(res.trim()).to.equal(
+            [
+                '@namespace "INDEX";',
```

Comment:
If we change it to an official generator, should the output be this concise, or further spaced apart?

Reply:
we will not expose it as a generator but I will expose the function that create the exports for everying